File: Silverlight/UpdateControls.Light.Demo.VB/Customer.vb
Public Class Customer
    Dim _name As String
    Dim _invoices As List(Of Invoice)

    Dim _indName As New Independent
    Dim _indInvoices As New Independent

    Public Property Name() As String
        Get
            _indName.OnGet()
            Return _name
        End Get
        Set(ByVal value As String)
            _indName.OnSet()
            _name = value
        End Set
    End Property

    Public ReadOnly Property Invoices() As IEnumerable(Of Invoice)
        Get
            _indInvoices.OnGet()
            Return _invoices
        End Get
    End Property

    Public Function NewInvoice(ByVal number As String) As Invoice
        _indInvoices.OnSet()
        Dim invoice As New Invoice(number)
        _invoices.Add(invoice)
        Return invoice
    End Function
End Class

File: Silverlight/UpdateControls.Light.Demo/Customer.cs
using System;
using System.Collections.Generic;
using UpdateControls.Collections;
using UpdateControls.Fields;

namespace UpdateControls.Light.Demo
{
    public class Customer
    {
        private Independent<string> _name = new Independent<string>();
        private IndependentList<Invoice> _invoices = new IndependentList<Invoice>();

        public string Name
        {
            get { return _name; }
            set { _name.Value = value; }
        }

        public IEnumerable<Invoice> Invoices
        {
            get { return _invoices; }
        }

        public Invoice NewInvoice(string number)
        {
            Invoice invoice = new Invoice(number);
            _invoices.Add(invoice);
            return invoice;
        }
    }
}
